plugins/pulp_docker/plugins/migrations/0006_add_arch_os_size_fields.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import os
+
+from pulp.server.db.connection import get_collection
+from pulp.server.db.migrations.lib import utils
+
+
+_logger = logging.getLogger('pulp_docjer.plugins.migrations.0006')
+
+
+def migrate(*args, **kwargs):
+    migrate_manifest_list()
+    migrate_blob()
+
+
+def migrate_manifest_list():
+    """
+    Update format of manifest on manifest lists to have digest, os, and arch fields
+    """
+
+    manifest_list_collection = get_collection('units_docker_manifest_list')
+    total_manifest_lists = manifest_list_collection.count(
+        {'manifests': {'$type': ["string"]}})
```

Comment:
this syntax will not work with older versions, we should stick to numbers for compatibilty

```
> db.units_docker_manifest_list.count({'manifests': {$type: ["string"]}})
2018-07-13T16:59:11.848+0200 E QUERY    [thread1] Error: count failed: {
	"waitedMS" : NumberLong(0),
	"ok" : 0,
	"errmsg" : "argument to $type is not a number or a string",
	"code" : 14
} :

> db.units_docker_manifest_list.count({'manifests': {'$type': 2}})
6

```

Reply:
I agree. the other option is to see if there are any plans of upgrading mongo.